timeseries/src/autogluon/timeseries/utils/warning_filters.py

```diff
@@ -70,3 +71,22 @@ def disable_stdout():
     sys.stdout = io.StringIO()
     yield
     sys.stdout = save_stdout
+
+
+class DuplicateLogFilter:
+    def __init__(self, max_count: int = 1):
+        self.messages = Counter()
+        self.max_count = max_count
+
+    def filter(self, record):
+        count = self.messages[record.msg]
+        self.messages[record.msg] += 1
+        return count < self.max_count
+
+
+@contextlib.contextmanager
+def disable_duplicate_logs(logger, max_count: int = 1):
```

Comment:
Where do the repeated log messages originate from?

Reply:
They come from the fact that predict_quantiles is called repeatedly by `ChronosModel`. 

Currently the `chronos` package takes care to only minimally change the model implementations from the upstream sources (in the `pipeline` package) and wrap them in `ChronosModel`. i.e., only `ChronosModel` is aware of its data loader which is not passed down to the pipeline implementations. This is an artifact of our choice to wrap pipeline implementations. However, this also results in this quirk: the bolt pipeline implementation will only know that it's being called with the wrong quantiles if it's going to be called multiple times with the same quantiles in the dataloader loop.